scripts/update_hook_revs.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Script to ensure the pre-commit hook revs match what is installed."""
+import re
+import sys
+from pathlib import Path
+
+REPO_ROOT_DIR = Path(__file__).parent.parent.resolve()
+PRE_COMMIT_CFG_PATH = REPO_ROOT_DIR / ".pre-commit-config.yaml"
+LOCK_FILE_PATH = REPO_ROOT_DIR / "requirements-dev.txt"
+
+
+def main():
+    """Compare configured pre-commit hooks with the installed dependencies. For the set
+    that overlap (e.g. `black`, `mypy`, `ruff`, etc.), make sure the versions match.
+    If running with `--check`, exit with status code 1 if anything is outdated.
+    If running without `--check`, update `.pre-commit-config.yaml` as needed.
+    Any invalid flags are ignored.
+    """
+    args = sys.argv[1:]
+    running_pre_commit = "--check" in args
```

Comment:
Please have a look at how we typically write the update scripts using our wrapper around typer:
https://github.com/ghga-de/microservice-repository-template/blob/main/scripts/update_lock.py#L185
https://github.com/ghga-de/microservice-repository-template/blob/main/scripts/update_lock.py#L257

We also have methods for reporting success or failures to the user:
https://github.com/ghga-de/microservice-repository-template/blob/main/scripts/update_lock.py#L248

Would be great if all of the scripts follow the same pattern.
https://github.com/ghga-de/microservice-repository-template/blob/main/scripts/update_lock.py#L248

Reply:
Main function split out and typer/cli.py used as of [13efb09](https://github.com/ghga-de/microservice-repository-template/pull/161/commits/13efb09e24bd7604d5d7b051809bd39c2061d8f2)